overlord/hookstate/context.go

```diff
@@ -165,7 +206,7 @@ func (c *Context) Get(key string, value interface{}) error {
 
```

Comment:
We need to make Get and Set work, as described above. If task == nil, we can work with a memory-only "hook-context" out of `c.cache`.